### PR TITLE
Update documentation for symfony4

### PR DESCRIPTION
**Pull request by kunicmarko20**
I am targeting this branch, because this is docs.

## Subject

I probably missed some references, but it should be :ok_hand:

### CODE REVIEW

File: docs/cookbook/recipe_custom_action.rst [hunk 1]
```diff
@@ -2,9 +2,14 @@ Creating a Custom Admin Action
 ==============================
 
 This is a full working example of creating a custom list action for SonataAdmin.
-The example is based on an existing ``CarAdmin`` class in an ``AppBundle``.
+The example is based on an existing ``CarAdmin`` class in an ``App``.
```
Comment (greg0ire) [inline docs/cookbook/recipe_custom_action.rst]: "in a an `App` namespace."

Comment (kunicmarko20) [reply docs/cookbook/recipe_custom_action.rst]: a an?

Comment (greg0ire) [reply docs/cookbook/recipe_custom_action.rst]: I'm tired :P

File: docs/cookbook/recipe_delete_field_group.rst [hunk 2]
```diff
@@ -22,24 +25,35 @@ You could delete every field by hand, using the ``FormMapper``s ``remove`` metho
         }
 
 
-This works, as long as the extended Admin does not use Groups to organize it's field. In the above example, we try to remove all fields from the User Admin, that comes with the SonataUserBundle.
-However, since the fields we deleted, are all part of the 'Social' Group of the form, the fields will be deleted and the empty group will stay.
-For this case, the FormMapper comes with a method, which allows you to get rid of a whole form group: ``removeGroup``
+This works, as long as the extended Admin does not use Groups to organize it's field.
```
Comment (greg0ire) [inline docs/cookbook/recipe_delete_field_group.rst]: "its" (not your fault I know :P)

File: docs/cookbook/recipe_knp_menu.rst [hunk 3]
```diff
@@ -97,15 +102,19 @@ And voilà, now you have a menu group which contains a link to a sonata admin vi
 Using a menu provider
 ---------------------
 
-As seen above, the main way to declare your menu is by declaring items in your sonata admin config file. In some case you may have to create a more complex menu depending on your business logic. This is possible by using a menu provider to populate a whole menu group. This is done with the ``provider`` config value.
+As seen above, the main way to declare your menu is by declaring items
+in your sonata admin config file. In some case you may have to create a
```
Comment (greg0ire) [inline docs/cookbook/recipe_knp_menu.rst]: "cases"

File: docs/cookbook/recipe_overwrite_admin_configuration.rst [hunk 4]
```diff
@@ -1,9 +1,12 @@
 Overwrite Admin Configuration
 =============================
 
-Sometime you might want to overwrite some Admin settings from vendors. This recipe will explain how to achieve this operation. However, keep in mind this operation is quite dangerous and might break code.
+Sometime you might want to overwrite some Admin settings from vendors.
```
Comment (greg0ire) [inline docs/cookbook/recipe_overwrite_admin_configuration.rst]: Sometimes

File: docs/reference/architecture.rst [hunk 5]
```diff
@@ -173,32 +178,32 @@ Your ``Admin`` class defines which of your model's fields will be available in e
...
         {
             $formMapper
-                ->add('title', 'text', [
+                ->add('title', TextType:class, [
                     'label' => 'Post Title'
                 ])
-                ->add('author', 'entity', [
-                    'class' => 'AppBundle\Entity\User'
+                ->add('author', EntityType::class, [
+                    'class' => 'App\Entity\User'
```
Comment (jordisala1991) [inline docs/reference/architecture.rst]: User::class

File: docs/reference/field_types.rst [hunk 6]
```diff
@@ -127,34 +129,36 @@ Parameter                               Description
...
 
+    use Symfony\Component\Form\Extension\Core\Type\UrlType;
+
     public function configureListFields(ListMapper $listMapper)
     {
         $listMapper
             // Output for value `http://example.com`:
             // `<a href="http://example.com">http://example.com</a>`
-            ->add('targetUrl', 'url')
+            ->add('targetUrl', UrlType::class)
```
Comment (jordisala1991) [inline docs/reference/field_types.rst]: This is a list mapper, so it does not use form types

File: docs/reference/field_types.rst [hunk 7]
```diff
@@ -86,11 +86,13 @@ The ``choice`` field type also supports multiple values that can be separated by
...
 .. code-block:: php
 
+    use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
     public function configureListFields(ListMapper $listMapper)
     {
         // For the value `['r', 'b']`, the displayed text ist `red | blue`.
         $listMapper
-            ->add('colors', 'choice', [
+            ->add('colors', ChoiceType::class, [
```
Comment (jordisala1991) [inline docs/reference/field_types.rst]: this is a list mapper, no form types

File: docs/reference/security.rst [hunk 8]
```diff
@@ -348,20 +350,14 @@ The following configuration for the SonataUserBundle defines:
             #sonata.admin.security.mask.builder:
```
Comment (jordisala1991) [inline docs/reference/security.rst]: Can you take a look at this PR and modify things here if neccessary? https://github.com/sonata-project/SonataAdminBundle/pull/4962

File: docs/cookbook/recipe_custom_action.rst [hunk 9]
```diff
@@ -46,30 +51,30 @@ Either by using XML:
...
+        <!-- src/Resources/config/admin.xml -->
 
-        <service id="app.admin.car" class="AppBundle\Admin\CarAdmin">
+        <service id="app.admin.car" class="App\Admin\CarAdmin">
             <tag name="sonata.admin" manager_type="orm" group="Demo" label="Car" />
             <argument />
-            <argument>AppBundle\Entity\Car</argument>
-            <argument>AppBundle:CRUD</argument>
+            <argument>App\Entity\Car</argument>
+            <argument>App:CRUD</argument>
```
Comment (Xymanek) [inline docs/cookbook/recipe_custom_action.rst]: This does not work:

```yaml
live2:
    path: /live2/{id}
    controller: App:Live:index
```

screenshot_3

Additionally it's also going to be deprecated in 4.1: https://symfony.com/blog/new-in-symfony-4-1-deprecated

Comment (kunicmarko20) [reply docs/cookbook/recipe_custom_action.rst]: That is service argument and not route configuration, but I got your point, just wrong example :smile:

**Comment by kunicmarko20** [comment]
I guess that should be it?

**Comment by jordisala1991** [comment]
Thank you @kunicmarko20

File: docs/cookbook/recipe_custom_action.rst (hunk 9 above)
Comment (Xymanek) [reply docs/cookbook/recipe_custom_action.rst]: Well it becomes route config later :) But yeah, had some brain lag - would've been better to showcase on an actual admin class

**Pull request merged**